File: 2017/d4/sol2.ex
File.read!("input.txt")
  |> String.trim
  |> String.split("\n", trim: true)
  |> Enum.reduce(0, fn(x, acc) ->
      pass = String.split(x)
          |> Enum.map(fn(x) ->
               String.codepoints(x)
               |> Enum.sort
               |> Enum.reduce(fn(x, acc) -> acc <> x end)
             end )
      if Enum.count(Enum.uniq(pass)) == Enum.count(pass) do
        acc + 1
      else
        acc
      end
    end)
  |> IO.puts

File: 2017/d5/sol2.rb
input = File.read("input.txt")

input = input.strip.split("\n").map { |e| e.to_i }

index = 0
jumps = 0

while index >= 0 && index < input.length
  val = input[index]
  jumps = jumps + 1
  input[index] = val >= 3 ? input[index] - 1 : input[index] + 1
  index = index + val
  # puts index
end

puts jumps
